components/ram.c

```diff
@@ -51,4 +51,77 @@ ram_used(void)
 	       bprintf("%f", (float)(total - free - buffers - cached) / 1024 / 1024) :
 	       NULL;
 }
+#elif defined(__OpenBSD__)
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+inline int
+load_uvmexp(struct uvmexp *uvmexp)
+{
+	int uvmexp_mib[] = {CTL_VM, VM_UVMEXP};
+	size_t size;
+
+	size = sizeof(*uvmexp);
+
+	return sysctl(uvmexp_mib, 2, uvmexp, &size, NULL, 0) >= 0 ? 1 : 0;
+}
+
+const char *
+ram_free(void)
+{
+	struct uvmexp uvmexp;
+	int free_pages, free;
+
+	if (load_uvmexp(&uvmexp)) {
+		free_pages = uvmexp.npages - uvmexp.active;
+		free = (free_pages * uvmexp.pagesize) >> 20;
+		return bprintf("%d", free);
+	}
+
+	return NULL;
+}
+
+const char *
+ram_perc(void)
+{
+	struct uvmexp uvmexp;
+	float percent;
```

Comment:
Please use `int` for percentage as in `__linux__`. Change function below appropriately.